webapp/javascript/pages/IntroPages/SignIn/index.tsx

```diff
@@ -0,0 +1,127 @@
+import { Link, useHistory } from 'react-router-dom';
+import cx from 'classnames';
+import Icon from '@ui/Icon';
+import { faGithub } from '@fortawesome/free-brands-svg-icons/faGithub';
+import InputField from '@ui/InputField';
+import StatusMessage from '@ui/StatusMessage';
+import { useAppDispatch, useAppSelector } from '@pyroscope/redux/hooks';
+import { logIn } from '@pyroscope/services/users';
+import {
+  loadCurrentUser,
+  selectCurrentUser,
+} from '@pyroscope/redux/reducers/user';
+import { GitlabIcon, GoogleIcon } from '../Icons';
+import Divider from '../Divider';
+import inputStyles from '../InputGroup.module.css';
+import styles from '../IntroPages.module.css';
+import buttonStyles from './buttons.module.css';
+
+function SignInPage() {
+  const history = useHistory();
+  const dispatch = useAppDispatch();
+  const currentUser = useAppSelector(selectCurrentUser);
+  const [form, setForm] = useState({
+    username: '',
+    password: '',
+    errors: [],
+  });
+
+  const handleFormChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const { name } = event.target;
+    const { value } = event.target;
+    setForm({ ...form, [name]: value });
+  };
+
+  async function handleSubmit(event: React.FormEvent<HTMLFormElement>) {
+    event.preventDefault();
+    try {
+      const { username, password } = {
+        ...form,
+      };
+
+      const res = await logIn({ username, password });
+      if (res.isOk) {
+        dispatch(loadCurrentUser());
+        return;
+      }
+
+      throw res.error;
+    } catch (e: any) {
+      setForm({ ...form, errors: e.errors || [e.message] });
+    }
+  }
+
+  useEffect(() => {
+    if (currentUser) {
+      history.push('/');
+    }
+  }, [currentUser]);
+
+  return (
+    <div className={styles.loginWrapper}>
+      <form className={styles.form} onSubmit={handleSubmit}>
+        <div className={styles.formHeader}>
+          <div className={styles.logo} />
+          <h1>Welcome to Pyroscope</h1>
+          <h3>Log in to continue</h3>
+        </div>
+        <div>
+          <StatusMessage type="error" message={form.errors?.join(', ')} />
+          <InputField
+            type="text"
+            name="username"
+            label="Username"
+            placeholder="Username"
+            className={inputStyles.inputGroup}
+            value={form.username}
+            onChange={handleFormChange}
+            required
+          />
+          <InputField
+            type="password"
+            name="password"
+            label="Password"
+            placeholder="Password"
+            className={inputStyles.inputGroup}
+            value={form.password}
+            onChange={handleFormChange}
+            required
+          />
+        </div>
+        <button className={styles.button} type="submit">
+          Log in
+        </button>
+        <Divider />
+        <div className={cx(buttonStyles.buttonContainer)}>
+          <Link
```

Comment:
In the original implementation these are added conditionally, only if these a particular sign in method is enabled. For example: https://github.com/pyroscope-io/enterprise/blob/97a60023e3e9d5b9fa705eacf0900386ee8367c7/webapp/templates/login.html#L50

We should probably pass these variables to the frontend the same way we pass other things for example, here's how we pass "features":
* https://github.com/pyroscope-io/pyroscope/blob/ac5a1284d5d9c925093a2be54deb720eff29bac8/pkg/server/handler.go#L110-L120
* https://github.com/pyroscope-io/enterprise/blob/97a60023e3e9d5b9fa705eacf0900386ee8367c7/webapp/templates/index.html#L39-L41
then later they are accessible via `window.features`



--- 
you can test all of this with a config like this:
```

auth:
  signup-default-role: Admin
  ingestion:
    enabled: true

  # Internal authentication provider settings.
  internal:
    enabled: true

  # Google OAuth2 authentication settings.
  google:
    enabled: false

  # Github OAuth2 authentication settings.
  github:
    enabled: true

  # Gitlab OAuth2 authentication settings.
  gitlab:
    enabled: false

```